test/library/input.spec.jsx

```diff
@@ -0,0 +1,1076 @@
+      });
+
+      // Replaces the test above
+      it('should correctly show the decimal values', async () => {
+        const { input, user, rerender } = await render(
+          <NumericFormat
+            value="123.123"
+            decimalScale={18}
+            thousandSeparator
+            prefix="$"
+            valueIsNumericString
+          />,
+        );
+
+        expect(input).toHaveValue('$123.123');
+
+        rerender(
+          <NumericFormat
+            value="123.1234"
+            decimalScale={18}
+            thousandSeparator
+            prefix="$"
+            valueIsNumericString
+          />,
+        );
+        expect(input).toHaveValue('$123.1234');
+      });
+
+      it.skip('should show the right number of zeros in all cases', async () => {
+        class WrapperComponent extends React.Component {
+          constructor() {
+            super();
+            this.state = {
+              value: '100.0',
+            };
+          }
+
+          render() {
+            return (
+              <NumericFormat
+                name="numberformat"
+                value={this.state.value}
+                thousandSeparator
+                prefix="$"
+                decimalScale={2}
+                valueIsNumericString
+              />
+            );
+          }
+        }
+
+        const { input, user } = await render(<WrapperComponent />);
+        expect(getInputValue(wrapper)).toEqual('$100.0');
+        wrapper.setState({ value: '123.00' });
+        expect(getInputValue(wrapper)).toEqual('$123.00');
+        wrapper.setState({ value: '132.000' });
+        expect(getInputValue(wrapper)).toEqual('$132.00');
+        wrapper.setState({ value: '100.10' });
+        expect(getInputValue(wrapper)).toEqual('$100.10');
+      });
+    });
+
+    // Replaces the test above
+    it('should show the correct number of zeroes after the decimal', async () => {
+      const { input, user, rerender } = await render(
+        <NumericFormat
+          decimalScale={2}
+          prefix="$"
+          thousandSeparator
+          value="100.0"
+          valueIsNumericString
+        />,
+      );
+
+      expect(input).toHaveValue('$100.0');
+
+      rerender(
+        <NumericFormat
+          decimalScale={2}
+          prefix="$"
+          thousandSeparator
+          value="123.00"
+          valueIsNumericString
+        />,
+      );
+
+      expect(input).toHaveValue('$123.00');
+
+      rerender(
+        <NumericFormat
+          decimalScale={2}
+          prefix="$"
+          thousandSeparator
+          value="132.000"
+          valueIsNumericString
+        />,
+      );
+
+      expect(input).toHaveValue('$132.00');
+
+      rerender(
+        <NumericFormat
+          decimalScale={2}
+          prefix="$"
+          thousandSeparator
+          value="100.10"
+          valueIsNumericString
+        />,
+      );
+
+      expect(input).toHaveValue('$100.10');
```

Comment:
**Suggestion**
When we rerendering so many times, we can create in place component/utils, to reduce the duplicats.
```suggestion
     
     const CustomNumericFormat = (props) => (
       <NumericFormat
          decimalScale={2}
          prefix="$"
          thousandSeparator
          valueIsNumericString
          {...props}
        />
     )

      const { input, user, rerender } = await render( <CustomNumericFormat value="100.0"/>);

      expect(input).toHaveValue('$100.0');

      rerender(<CustomNumericFormat  value="123.00"  />);

      expect(input).toHaveValue('$123.00');

      rerender(<CustomNumericFormat value="132.000" />);

      expect(input).toHaveValue('$132.00');

      rerender(<CustomNumericFormat value="100.10" />);

      expect(input).toHaveValue('$100.10');
```

Reply:
The plan here is to split tests like this into smaller tests, but point taken.